platform/osx/display_server_osx.mm

```diff
@@ -3374,6 +3374,8 @@ void _update_keyboard_layouts() {
 }
 
 DisplayServer::WindowID DisplayServerOSX::get_window_at_screen_position(const Point2i &p_position) const {
+#warning This is an incorrect implementation, if windows overlap, it should return the topmost visible one or none if occluded by a foreign window
```

Comment:
```obj-c++
	Point2i position = p_position;
	position.y *= -1;
	position += _get_screens_origin();
	position /= screen_get_max_scale();

	NSInteger wnum = [NSWindow windowNumberAtPoint:NSMakePoint(position.x, position.y) belowWindowWithWindowNumber:0 /*topmost*/];
	for (Map<WindowID, WindowData>::Element *E = windows.front(); E; E = E->next()) {
		if ([E->get().window_object windowNumber] == wnum) {
			return E->key();
		}
	}
	return INVALID_WINDOW_ID;
```

Reply:
I suggested that @reduz amends this in this commit but he forgot, so let's merge and you can make a fixup PR.

Reply:
ah I forgot about this, please feel free to make it a separate PR

Reply:
I have not removed it from #40084 when rebasing it after this PR, so relevant code already get merged with it.